src/Admin/Permissions.vue

```diff
@@ -144,9 +220,99 @@ export default {
       }
       else {
         await this.$agent.p_mfa('ws_db_permission', 'unset', [this.id, perm])
-        delete this.effective[perm]
+        this.effective[perm] = false
       }
-    }
+    },
+    tabsVisible: function (perm) {
+      if (this.ui == 'supervisor') {
+        return (perm != 'admin-ui' && perm != 'profile-ui')
+      }
+      else if (this.ui == 'agent') {
+        return (perm == 'main-ui' || perm == 'profile-ui')
+      }
+      else {
+        return true
+      }
+    },
+    isAgent () {
+      return this.ui == 'agent'
+    },
+    profileSelected () {
+      return (this.effective['profile-ui'] && this.isAgent())
+    },
+    selectAll () {
+      let self = this
+      Object.keys(this.names).forEach(function (key) {
+        self.onChange(key, true)
+      })
+    },
+    unselectAll () {
+      let self = this
+      Object.keys(this.names).forEach(function (key) {
+        self.onChange(key, false)
+      })
+
+    },
+    selectTabs () {
```

Comment:
btw, I would prefer two have two functions, select/unselect and just pass feature group name as a parameter

Reply:
i will change it... it makes sense to have only 2 methods